fix: RankingFloatingActionButton partially off-screen for some translations

Zutzsa opened the pull request:
### What
<!-- description of the PR -->
- Wrapped the RankingFloatingActionButton inside an Expanded widget and its child inside a FittedBox widget.
<!-- 
Please name your pull request following this scheme: `type: What you did` this allows us to automatically generate the changelog
Following `type`s are allowed:
  - `feat`, for Features
  - `fix`, for Bug Fixes
  - `docs`, for Documentation
  - `ci`, for Automation
  - `refactor`, for code Refactoring
  - `chore`, for Miscellaneous things
-->
### Screenshot

After:

https://github.com/openfoodfacts/smooth-app/assets/16384897/edd17cb5-3ea7-4172-bb7d-03520d7d3970



### Fixes bug(s)
- [5116](https://github.com/openfoodfacts/smooth-app/issues/5116)

Commit:
Bug: wrapping RankingFloatingActionButton inside a Expanded widget and it's childs inside a FittedBox widget

Commit:
Merge branch 'develop' into bug/ranking_float_action_button_overflow

monsieurtanuki reviewed (commented):
Hi @gmmoraes!

To be honest I'm not 100% convinced by your fix :(

First: in the case of this specific issue/PR, screenshots would be a better option than video, if we want to check the alignments and the sizes of buttons.

Second: we already use `AutoSizeText` in other parts of the code, so that texts fit in a box. I would recommend using it, with `maxLines: 2` for instance.

I would also summarize what we try to achieve here:
* "top" behavior: a "floating action button like" button with an icon and a text that takes all the screen width (minus padding)
* "bottom" behavior
  * a row of the same height and width as in top scenario
  * with an "icon+text" button the same height as before, with an `AutoSizeText`, that takes as much space as possible, width-wise
  * then some padding
  * then the up arrow button

Zutzsa commented:
> Hi @gmmoraes!
> 
> To be honest I'm not 100% convinced by your fix :(
> 
> First: in the case of this specific issue/PR, screenshots would be a better option than video, if we want to check the alignments and the sizes of buttons.
> 
> Second: we already use `AutoSizeText` in other parts of the code, so that texts fit in a box. I would recommend using it, with `maxLines: 2` for instance.
> 
> I would also summarize what we try to achieve here:
> 
> * "top" behavior: a "floating action button like" button with an icon and a text that takes all the screen width (minus padding)
> * "bottom" behavior
>   
>   * a row of the same height and width as in top scenario
>   * with an "icon+text" button the same height as before, with an `AutoSizeText`, that takes as much space as possible, width-wise
>   * then some padding
>   * then the up arrow button

Hey @monsieurtanuki !

Thank you for the feedback!

I will take a look at the AutoSizeText package, and later this week, I will try to refactor the PR code following these instructions.

Commit:
fix: Adding AutoSizeText to FloatingActionButton and adding it to a FittedBox

Commit:
fix: returning RankingFloatingActionButton class comments

Zutzsa commented:
Hey @monsieurtanuki !

I've pushed a new commit. Please let me know if it's okay or if it requires any changes.

I've added `AutoSizeText` as the `label` property of `RankingFloatingActionButton`. 
My main difficulty was managing a way for `RankingFloatingActionButton`'s width constraints to affect its child widgets (more specifically `AutoSizeText`), because `FloatingActionButton`, and consequently its text, were not being resized despite `RankingFloatingActionButton`'s width. To address this, I added `RankingFloatingActionButton` inside a `Flexible` widget so that `Flexible` can pass down its constraints. Additionally, I used a `FittedBox` widget to manage the resizing of child widgets.

Thanks in advance.

monsieurtanuki commented:
Hi @gmmoraes!
Could you please attach 4 screenshots:
* a full width FAB with a fake short text ("Hello")
* a full width FAB with a fake 2-line text ("This is a test text that would fit in 2 lines bla bla bla")
* a FAB on the left with a fake short text, and the icon button on the right
* a full width FAB with a fake 2-line text, and the icon button on the right

Zutzsa commented:
@monsieurtanuki , I am going send this weekend without fail!

Zutzsa commented:
Hey @monsieurtanuki!

After running the tests, I am not sure if the current code solves this problem. Instead of breaking to a new line, it is reducing the font size.

Is it necessary to use `FloatingActionButton` in this widget? I am not sure why, but multiline is not working with it. However, by using `ElevatedButton`, I was able to achieve it.

With that being said, I have added below pictures of the current commit and pictures of using `ElevatedButton` instead of `FloatingActionButton`. Please let me know what you think.


### Current commit Screen




<details>

- a full width FAB with a fake short text ("Hello")

![Screenshot 2024-04-07 at 18 58 25](https://github.com/openfoodfacts/smooth-app/assets/16384897/9622fb6c-3535-4772-9a1b-c875eb00ac3e)

- a full width FAB with a fake 2-line text ("This is a test text that would fit in 2 lines bla bla bla")

![Screenshot 2024-04-07 at 18 59 59](https://github.com/openfoodfacts/smooth-app/assets/16384897/d5a35e1f-3d1b-4876-b835-89a8f49eba07)


- a FAB on the left with a fake short text, and the icon button on the right
![Screenshot 2024-04-07 at 18 58 38](https://github.com/openfoodfacts/smooth-app/assets/16384897/3187fdc5-1148-4210-b718-2a798f47417a)

- a full width FAB with a fake 2-line text, and the icon button on the right

![Screenshot 2024-04-07 at 19 00 07](https://github.com/openfoodfacts/smooth-app/assets/16384897/c0834a07-ba26-4dc3-8a3f-d9d839fb73f4)


</details>



### With ElevatedButton

<details>

- a full width FAB with a fake short text ("Hello")

![Screenshot 2024-04-07 at 19 33 26](https://github.com/openfoodfacts/smooth-app/assets/16384897/50fcfc9e-456c-4d82-a403-72fb5d25d56c)

- a full width FAB with a fake 2-line text ("This is a test text that would fit in 2 lines bla bla bla")

![Screenshot 2024-04-07 at 19 29 11](https://github.com/openfoodfacts/smooth-app/assets/16384897/f3816ad4-004e-4b59-8f6f-ccd27247c8df)


- a FAB on the left with a fake short text, and the icon button on the right

![Screenshot 2024-04-07 at 19 34 59](https://github.com/openfoodfacts/smooth-app/assets/16384897/0780e06d-64b1-494d-ad21-33eb7bfe4d0b)



- a full width FAB with a fake 2-line text, and the icon button on the right

![Screenshot 2024-04-07 at 19 29 20](https://github.com/openfoodfacts/smooth-app/assets/16384897/421f3d04-c0b0-4ef0-8c01-a30b1f72b4da)


</details>

monsieurtanuki commented:
Hi @gmmoraes! Thank you for your screenshots!

If I put the versions side by side, it's obvious that the `ElevatedButton` versions look better. And as we're not using the `FloatingActionButton` in a standard way anyway (because of the other "to the top" button), we may as well use `ElevatedButton` indeed. I'm not 100% satisfied with the height and the position, though.
| current commit | ElevatedButton |
| -- | -- |
| ![Screenshot 2024-04-07 at 18 58 25](https://github.com/openfoodfacts/smooth-app/assets/16384897/9622fb6c-3535-4772-9a1b-c875eb00ac3e) | ![Screenshot 2024-04-07 at 19 33 26](https://github.com/openfoodfacts/smooth-app/assets/16384897/50fcfc9e-456c-4d82-a403-72fb5d25d56c) |
| ![Screenshot 2024-04-07 at 18 59 59](https://github.com/openfoodfacts/smooth-app/assets/16384897/d5a35e1f-3d1b-4876-b835-89a8f49eba07) | ![Screenshot 2024-04-07 at 19 29 11](https://github.com/openfoodfacts/smooth-app/assets/16384897/f3816ad4-004e-4b59-8f6f-ccd27247c8df) |
| ![Screenshot 2024-04-07 at 18 58 38](https://github.com/openfoodfacts/smooth-app/assets/16384897/3187fdc5-1148-4210-b718-2a798f47417a) | ![Screenshot 2024-04-07 at 19 34 59](https://github.com/openfoodfacts/smooth-app/assets/16384897/0780e06d-64b1-494d-ad21-33eb7bfe4d0b) |
| ![Screenshot 2024-04-07 at 19 00 07](https://github.com/openfoodfacts/smooth-app/assets/16384897/c0834a07-ba26-4dc3-8a3f-d9d839fb73f4) | ![Screenshot 2024-04-07 at 19 29 20](https://github.com/openfoodfacts/smooth-app/assets/16384897/421f3d04-c0b0-4ef0-8c01-a30b1f72b4da) |

Here are my suggestions to display relevant heights:
* use `ElevatedButton` for `RankingFloatingActionButton`, with a fixed height of MINIMUM_TOUCH_SIZE
* use `ElevatedButton` also for the "to the top" arrow button in `product_query_page.dart`, with a fixed height of MINIMUM_TOUCH_SIZE

Here are my suggestions to display consistent layouts:
* align `RankingFloatingActionButton` to the right when alone, instead of centering it
* align `RankingFloatingActionButton` then the arrow button to the right, in order to avoid the space between both buttons

@gmmoraes What do you think of that?

monsieurtanuki commented:
@gmmoraes Still working on it?

Zutzsa commented:
@monsieurtanuki  , Apologies for the delayed response; I wasn't able to work on this task for the past month.

Regarding the suggestion, absolutely, it makes sense! Let's implement it as you've outlined. Thanks for the input!

I will try to implement the required changes this week. Next week, It will also begin my 10 days of holidays, so in a worst-case scenario, I will have a lot of time to work on this.

Does that sound okay?

monsieurtanuki commented:
@gmmoraes No problem, take your time!

Zutzsa commented:
> Hi @gmmoraes! Thank you for your screenshots!
> 
> If I put the versions side by side, it's obvious that the `ElevatedButton` versions look better. And as we're not using the `FloatingActionButton` in a standard way anyway (because of the other "to the top" button), we may as well use `ElevatedButton` indeed. I'm not 100% satisfied with the height and the position, though.
> 
> current commit	ElevatedButton
> ![Screenshot 2024-04-07 at 18 58 25](https://private-user-images.githubusercontent.com/16384897/320320509-9622fb6c-3535-4772-9a1b-c875eb00ac3e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwODU2NzYsIm5iZiI6MTcxNTA4NTM3NiwicGF0aCI6Ii8xNjM4NDg5Ny8zMjAzMjA1MDktOTYyMmZiNmMtMzUzNS00NzcyLTlhMWItYzg3NWViMDBhYzNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA3VDEyMzYxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxNzkyYzAxNzU5N2VlYzc5MmY2MjUwYjE3ZDJiMjVhN2ViNzVjMDVkYzNjMjU5NDVhNDMxM2ZiOWQ3OTg4MWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xvpMfnLENuqwzjv6IxyyWW5vcexsZ2G_XPq87bN9vew)	![Screenshot 2024-04-07 at 19 33 26](https://private-user-images.githubusercontent.com/16384897/320320810-50fcfc9e-456c-4d82-a403-72fb5d25d56c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwODU2NzYsIm5iZiI6MTcxNTA4NTM3NiwicGF0aCI6Ii8xNjM4NDg5Ny8zMjAzMjA4MTAtNTBmY2ZjOWUtNDU2Yy00ZDgyLWE0MDMtNzJmYjVkMjVkNTZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA3VDEyMzYxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBhMGE0MzY4YjAwNjZmM2VjNmI4NDgzNjNiMmExODljYzM1OTE3YjNjMzI2Mzk4N2E5YzE3NTgwNTAwZmRlYTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.haGM7VhQTfPvJlPk_lK8fEfOsGTqwd-e4FkwKkoBR7s)
> ![Screenshot 2024-04-07 at 18 59 59](https://private-user-images.githubusercontent.com/16384897/320320696-d5a35e1f-3d1b-4876-b835-89a8f49eba07.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwODU2NzYsIm5iZiI6MTcxNTA4NTM3NiwicGF0aCI6Ii8xNjM4NDg5Ny8zMjAzMjA2OTYtZDVhMzVlMWYtM2QxYi00ODc2LWI4MzUtODlhOGY0OWViYTA3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA3VDEyMzYxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxNjBkZDQ0ZDMxNGZiZmZhYzdmNmQwMjU3OWE4ZjU0MjBiZTEwYzEwNzU1NGM5YzQxMjdlZTVkNDk5ZTNjMzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OoKoKWZ04CEUTkZ9mKj_aF7FnnnBpuGdbIjppT5FCT8)	![Screenshot 2024-04-07 at 19 29 11](https://private-user-images.githubusercontent.com/16384897/320320990-f3816ad4-004e-4b59-8f6f-ccd27247c8df.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwODU2NzYsIm5iZiI6MTcxNTA4NTM3NiwicGF0aCI6Ii8xNjM4NDg5Ny8zMjAzMjA5OTAtZjM4MTZhZDQtMDA0ZS00YjU5LThmNmYtY2NkMjcyNDdjOGRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA3VDEyMzYxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ1MjU1YzBjOGU1Y2NjMTc3OTAxM2I3YWZmOGZmZjU0YzEyNjkzMTZiYTdmZGM0OTE3NGZmYjFhN2QyZjI4M2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NTVORhpQ3DvbkCwjrSrR1fskk4QvlkMEcWN8iHXcT9c)
> ![Screenshot 2024-04-07 at 18 58 38](https://private-user-images.githubusercontent.com/16384897/320320552-3187fdc5-1148-4210-b718-2a798f47417a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwODU2NzYsIm5iZiI6MTcxNTA4NTM3NiwicGF0aCI6Ii8xNjM4NDg5Ny8zMjAzMjA1NTItMzE4N2ZkYzUtMTE0OC00MjEwLWI3MTgtMmE3OThmNDc0MTdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA3VDEyMzYxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg2YTc4Mjk0MTQwYzhlYWNhYWU1NDhhNjhmMmNlNmU5ZTQ5MDA4MzQxMjI2Yzk1YjhkOWYwMjBkZDUwNjA3NjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Xq0D8imWMtxyqn4OK0ttzM1YE8dkuAgEeFtH6R3w5_g)	![Screenshot 2024-04-07 at 19 34 59](https://private-user-images.githubusercontent.com/16384897/320321008-0780e06d-64b1-494d-ad21-33eb7bfe4d0b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwODU2NzYsIm5iZiI6MTcxNTA4NTM3NiwicGF0aCI6Ii8xNjM4NDg5Ny8zMjAzMjEwMDgtMDc4MGUwNmQtNjRiMS00OTRkLWFkMjEtMzNlYjdiZmU0ZDBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA3VDEyMzYxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThmZWUyY2ZjYTZhZGM4YmJiY2E2ZjRiOTNmYjg0MzE1NWVlYzg1YTJjZTk0NzVkYmRmMzE0NTNkZDM0ZWQ5M2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.w-2qBFj_p_vSzoStyt6l3Qczt8Ju9tKHmtyiRiIhd7E)
> ![Screenshot 2024-04-07 at 19 00 07](https://private-user-images.githubusercontent.com/16384897/320320655-c0834a07-ba26-4dc3-8a3f-d9d839fb73f4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwODU2NzYsIm5iZiI6MTcxNTA4NTM3NiwicGF0aCI6Ii8xNjM4NDg5Ny8zMjAzMjA2NTUtYzA4MzRhMDctYmEyNi00ZGMzLThhM2YtZDlkODM5ZmI3M2Y0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA3VDEyMzYxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFiM2Y4YzEwOGQ4YzIyZDM0NDU1ZjAyOTVjY2NhY2E5NTczZTBmNmQ0YWFlZTQwZDc2NmMyOTg1NTJjMjkxNTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.mJR2Y9Qh-C-ikYhiJ1gwVfexmStqR4C8kLbt3k2k3xg)	![Screenshot 2024-04-07 at 19 29 20](https://private-user-images.githubusercontent.com/16384897/320321016-421f3d04-c0b0-4ef0-8c01-a30b1f72b4da.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwODU2NzYsIm5iZiI6MTcxNTA4NTM3NiwicGF0aCI6Ii8xNjM4NDg5Ny8zMjAzMjEwMTYtNDIxZjNkMDQtYzBiMC00ZWYwLThjMDEtYTMwYjFmNzJiNGRhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA3VDEyMzYxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI1ODJhOWVkOWU5YmZkYzM2OTM1ZWJhMTQwYzExOTZiNDMyYjA5NmZhMzU2YjYxYjU0ZTVmMmNjNGE2ZTE1ODImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.sl00LwVibin5rq8KprWFfvvtgAO9Erov-KJlApdw9Pc)
> Here are my suggestions to display relevant heights:
> 
> * use `ElevatedButton` for `RankingFloatingActionButton`, with a fixed height of MINIMUM_TOUCH_SIZE
> * use `ElevatedButton` also for the "to the top" arrow button in `product_query_page.dart`, with a fixed height of MINIMUM_TOUCH_SIZE
> 
> Here are my suggestions to display consistent layouts:
> 
> * align `RankingFloatingActionButton` to the right when alone, instead of centering it
> * align `RankingFloatingActionButton` then the arrow button to the right, in order to avoid the space between both buttons
> 
> @gmmoraes What do you think of that?

@monsieurtanuki, regarding the previous suggestions, if you don't mind, I would like to confirm two pieces of information:

Concerning the `RankingFloatingActionButton` and `BackToTopButton` height issue from my previous images, was the problem that the `BackToTopButton` had a larger height?

Regarding the alignment requirement, should there be no horizontal space between the `RankingFloatingActionButton` and the `BackToTopButton`?

![Screenshot 2024-05-07 at 12 04 38](https://github.com/openfoodfacts/smooth-app/assets/16384897/2008dff1-6736-4fdc-996a-b5e1cd54f374)

monsieurtanuki commented:
@gmmoraes Your latest screenshot looks ok to me:
* same height for both buttons, which brings uniformity to the UI
* a space between both buttons
* a padding to the left of the left and the right of the right
* a text that is automatically resized as it is too big

Commit:
refactor: Replacing FloatingActionButton for ElevatedButton inside RankingFloatingActionButton

Commit:
fix: Limiting BackToTopButton height to MINIMUM_TOUCH_SIZE and adjusting alignment logic

Zutzsa committed:
Merge branch 'openfoodfacts:develop' into bug/ranking_float_action_button_overflow

Zutzsa commented:
@monsieurtanuki , I've pushed a new commit. Please let me know if it's okay or if it requires any changes.

- Implemented `ElevatedButton` with a fixed height of `MINIMUM_TOUCH_SIZE` inside `RankingFloatingActionButton`.
- Changed `BackToTopButton` to have a fixed height of `MINIMUM_TOUCH_SIZE`.
- Modified alignment logic for `RankingFloatingActionButton` inside `product_query_page.dart`, ensuring it aligns to the right when the `BackToTopButton` is not visible.

Cheers

monsieurtanuki commented on packages/smooth_app/lib/pages/product/common/product_query_page.dart:
```diff
@@ -157,19 +157,21 @@ class _ProductQueryPageState extends State<ProductQueryPage>
-                ? MainAxisAlignment.spaceBetween
-                : MainAxisAlignment.center,
+                ? MainAxisAlignment.end
+                : MainAxisAlignment.end,
```
In that case the test doesn't make sense anymore, does it?
It's always `.end`, right?

monsieurtanuki commented on packages/smooth_app/lib/pages/product/common/product_query_page.dart:
```diff
@@ -187,15 +189,24 @@ class _ProductQueryPageState extends State<ProductQueryPage>
-                          color: themeData.colorScheme.onSecondary,
+                      child: SizedBox(
+                        height: MINIMUM_TOUCH_SIZE,
+                        child: ElevatedButton(
```
I guess `IconButton` would make more sense and the code would be less verbose.
Would you please try it?

Zutzsa replied:
> I guess `IconButton` would make more sense and the code would be less verbose. Would you please try it?

If we do use the `IconButton`, I believe it will need to be embedded inside an `InkResponse` widget for a circular ripple effect and inside a `Material` widget for background and elevation.

Perhaps something like this:

 ```
     ClipOval(
        child: Material(
          elevation: 2.0,
          color: themeData.colorScheme.secondary,
          child: InkResponse(
            containedInkWell: true,
            child: SizedBox(
              height: MINIMUM_TOUCH_SIZE,
              child: IconButton(
                icon: const Icon(
                  Icons.arrow_upward,
                ),
                color: themeData.colorScheme.onSecondary,
                onPressed: onPressed,
              ),
            ),
          ),
        ),
      );
```

Perhaps it could be added inside a private method that returns a widget:

```
Widget _getBackToTopButton(ThemeData themeData, void Function()? onPressed) =>
      ClipOval(
        child: Material(
          elevation: 2.0,
          color: themeData.colorScheme.secondary,
          child: InkResponse(
            containedInkWell: true,
            child: SizedBox(
              height: MINIMUM_TOUCH_SIZE,
              child: IconButton(
                icon: const Icon(
                  Icons.arrow_upward,
                ),
                color: themeData.colorScheme.onSecondary,
                onPressed: onPressed,
              ),
            ),
          ),
        ),
      );
```

What do you think?

monsieurtanuki replied:
@gmmoraes My assumption was that it was easy to use (and to maintain) an `IconButton` in order to display a clickable disk with an icon inside. Easier than an `ElevatedButton`.
If it's not the case, don't waste time with the hard to read/maintain solution you've just suggested, and ignore my suggestion about using an `IconButton`.

monsieurtanuki commented on packages/smooth_app/lib/pages/product/common/product_query_page.dart:
```diff
@@ -157,19 +157,21 @@ class _ProductQueryPageState extends State<ProductQueryPage>
-                        barcodes: _model.displayBarcodes,
-                        title: widget.name,
+              Expanded(
+                child: Padding(
```
The `padding` is pointless here and probably causes confusion. Please try to remove it.

monsieurtanuki commented on packages/smooth_app/lib/widgets/ranking_floating_action_button.dart:
```diff
@@ -24,13 +24,27 @@ class RankingFloatingActionButton extends StatelessWidget {
+                  style: ButtonStyle(
+                    shape: MaterialStateProperty.all<RoundedRectangleBorder>(
+                      const RoundedRectangleBorder(
+                          borderRadius: BorderRadius.all(Radius.circular(40))),
```
```suggestion
                          borderRadius: CIRCULAR_RADIUS,
```

monsieurtanuki commented on packages/smooth_app/lib/widgets/ranking_floating_action_button.dart:
```diff
@@ -24,13 +24,27 @@ class RankingFloatingActionButton extends StatelessWidget {
-              label: Text(AppLocalizations.of(context).myPersonalizedRanking),
-              onPressed: onPressed,
-            ),
+            Flexible(
```
Generally speaking I'm not sure we need that `Row` and especially that `SizedBox`.

monsieurtanuki reviewed (commented):
Thank you @gmmoraes for your latest version! I guess we're very close to the solution.
Please have a look at my comments, that are suggestions about simplifying the code.
Please also attach screenshots in the 4 interesting cases:
1. big label, no arrow button
2. big label, arrow button
1. small label, no arrow button
2. small label, arrow button

Zutzsa commented:
@monsieurtanuki , 
Thank you for your feedback! I'll review the code and work on the suggestions. 
As for the screenshots, agreed, I will attach them in my next comment for the following version.

Cheers,

Commit:
fix: Changed the  property of a widget from conditional to fixed value and removed redundant Padding

Commit:
fix: Refactor widget structure from Row to Container

Zutzsa committed:
Merge branch 'openfoodfacts:develop' into bug/ranking_float_action_button_overflow

Zutzsa commented:
@monsieurtanuki , I've pushed a new commit. Please let me know if it's okay or if it requires any changes.
Below, I have added the requested images.

Edit: Each text below is a toggle button, to show the image just need to click on it.

<details>
<summary>big label, no arrow button</summary>

![Screenshot 2024-05-13 at 15 01 59](https://github.com/openfoodfacts/smooth-app/assets/16384897/54dbc976-4539-4aca-9f98-86b77ff2c298)

</details>


<details>
<summary>big label, arrow button</summary>

![Screenshot 2024-05-13 at 15 01 22](https://github.com/openfoodfacts/smooth-app/assets/16384897/4caafc78-0732-42a2-bd7a-2f6724b9681e)

</details>

<details>
<summary>medium label, no arrow button</summary>

![Screenshot 2024-05-13 at 15 03 07](https://github.com/openfoodfacts/smooth-app/assets/16384897/80dda9d9-d983-4af6-adc6-fa176870a444)


</details>

<details>
<summary>medium label, arrow button</summary>

![Screenshot 2024-05-13 at 15 03 17](https://github.com/openfoodfacts/smooth-app/assets/16384897/352b5bc6-c01f-488c-9f6e-4f8c8db55f0d)



</details>

<details>
<summary>small label, no arrow button</summary>

![Screenshot 2024-05-13 at 15 04 06](https://github.com/openfoodfacts/smooth-app/assets/16384897/190d77c5-eedc-4bda-a481-c643b01d26ef)

</details>

<details>
<summary>small label, arrow button</summary>

![Screenshot 2024-05-13 at 15 03 55](https://github.com/openfoodfacts/smooth-app/assets/16384897/044a1225-7de9-46e9-9d61-427f757731bf)

</details>

Cheers!

monsieurtanuki reviewed (approved):
Thank you @gmmoraes for the good work!
Perhaps there will be minor cosmetic changes later, but now we have a stable code that manages typical cases with different text lengths.
